hbase-server/src/test/java/org/apache/hadoop/hbase/io/hfile/TestCacheOnWrite.java

```diff
@@ -448,16 +455,34 @@ private void testNotCachingDataBlocksDuringCompactionInternals(boolean useTags)
       }
       region.flush(true);
     }
+
     clearBlockCache(blockCache);
     assertEquals(0, blockCache.getBlockCount());
+
     region.compact(false);
     LOG.debug("compactStores() returned");
 
+    boolean dataBlockCached = false;
     for (CachedBlock block: blockCache) {
-      assertNotEquals(BlockType.ENCODED_DATA, block.getBlockType());
-      assertNotEquals(BlockType.DATA, block.getBlockType());
+      if (BlockType.ENCODED_DATA.equals(block.getBlockType())
+          || BlockType.DATA.equals(block.getBlockType())) {
+        dataBlockCached = true;
+        break;
+      }
     }
+
+    // Data blocks should be cached in instances where we are caching blocks on write. In the case
+    // of testing
+    // BucketCache, we cannot verify block type as it is not stored in the cache.
```

Comment:
Why? If BucketCache is not giving the block type correctly, then its a bug there IMO.

Reply:
Seems it is only the iterator issue - that is being used here.